tests/tasks/upgrade_and_assert.yml

```diff
@@ -27,10 +27,17 @@
       fail:
         msg: The above task must fail
   rescue:
-    # The error is returned from a loop, as a result ansible_failed_result has
-    # a list of results of each loop invocation under
-    # ansible_failed_result.results and it is hard to reach the exact error
-    # message, hence using a general ansible_failed_result.msg here
+    # Verify the error message
+    # Verify that the error is returned only once in ansible_failed_result
     - name: Assert that the role failed with the correct message
       assert:
-        that: "'One or more items failed' in ansible_failed_result.msg"
+        that:
+          - >-
+            'You set mssql_version to {{ __mssql_prev_version }}, but your SQL'
+            in ansible_failed_result.results
+            | selectattr('msg', 'defined')
+            | map(attribute='msg')
+            | join(' ')
+          - ansible_failed_result.results | selectattr('msg', 'defined')
```

Comment:
If you have this condition first, then can omit the `selectattr('msg', 'defined')` in the second condition

Reply:
Hm, removing selectattr didn't work for me. selectattr goes into the list where the msg dict is defined, so that map than can map from it.

Reply:
That's how ansible_failed_result looks like in my case:
```
    Thursday 15 December 2022  21:59:32 +0100 (0:00:00.064)       0:02:08.874 *****
    ok: [/home/spetrosi/.cache/linux-system-roles/centos-7.qcow2c.snap] => {
        "ansible_failed_result": {
            "changed": false,
            "failed": true,
            "msg": "All items completed",
            "results": [
                {
                    "ansible_loop_var": "item",
                    "changed": false,
                    "failed": true,
                    "item": {
                        "key": 2017,
                        "value": 14
                    },
                    "msg": [
                        "\"You set mssql_version to 2017, but your SQL Server is of higher version\"",
                        "The role does not support downgrading SQL Server"
                    ]
                },
                {
                    "ansible_loop_var": "item",
                    "changed": false,
                    "item": {
                        "key": 2019,
                        "value": 15
                    },
                    "skip_reason": "Conditional result was False",
                    "skipped": true
                },
                {
                    "ansible_loop_var": "item",
                    "changed": false,
                    "item": {
                        "key": 2022,
                        "value": 16
                    },
                    "skip_reason": "Conditional result was False",
                    "skipped": true
                }
            ]
        }
    }
```

Reply:
ok - then please proceed with your solution